### PR TITLE
fix(plugin-meetings): consider undefined return type

### PR DESCRIPTION
**Pull request by shnaaz**
# COMPLETES #< INSERT LINK TO ISSUE >

## This pull request addresses

< DESCRIBE THE CONTEXT OF THE ISSUE >

## by making the following changes

< DESCRIBE YOUR CHANGES >
getPermissionTokenExpiryInfo returns undefined as well, this PR makes sure there is protection in code for that.
We epect typescript to complain but in sdk repo a function returning undefined is considered as having return type of 'any' hence typing getPermissionTokenExpiryInfo didn't help. Will raise a jira so that we can identify the cause of that, could be some misconfiguration

### Change Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Tooling change
- [ ] Internal code refactor

## The following scenarios where tested

< ENUMERATE TESTS PERFORMED, WHETHER MANUAL OR AUTOMATED >

### I certified that

- [ ] I have read and followed [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request)
- [ ] I discussed changes with code owners prior to submitting this pull request

- [ ] I have not skipped any automated checks
- [ ] All existing and new tests passed
- [ ] I have updated the documentation accordingly

---

Make sure to have followed the [contributing guidelines](https://github.com/webex/webex-js-sdk/blob/master/CONTRIBUTING.md#submitting-a-pull-request) before submitting.

### CODE REVIEW

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 1]
```diff
@@ -1525,7 +1525,9 @@ export default class Meeting extends StatelessWebexPlugin {
       : this.destination;
     const destinationType = isStartingSpaceInstantV2Meeting ? _MEETING_LINK_ : this.destinationType;
 
-    const {timeLeft, expiryTime, currentTime} = this.getPermissionTokenExpiryInfo();
+    const timeLeft = this.getPermissionTokenExpiryInfo()?.timeLeft;
+    const expiryTime = this.getPermissionTokenExpiryInfo()?.expiryTime;
```
Comment (Coread) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:1529]: suggestion: call the function once and extract after

File: packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js [hunk 2]
```diff
@@ -9795,7 +9840,7 @@ describe('plugin-meetings', () => {
```
Comment (nataliadelmar) [inline packages/@webex/plugin-meetings/test/unit/spec/meeting/index.js:9840]: we should add a text to see if nothing breaks if we do not get a numberable string as the exp. For example, 
- if we get 'woops' as the exp. Number('woops') will return NaN. How should the function behave then?
- if we get '{something: 'hello'} as the exp. Number(Object) will return NaN too. How should the function behave then?

**Review by nataliadelmar — APPROVED** [review]

**Review by torgeadelin — APPROVED** [review]

**Review by robstax — APPROVED** [review]

**Comment by torgeadelin** [comment]
i think we need a 5th one ::))))))

File: packages/@webex/plugin-meetings/src/meeting/index.ts [hunk 3]
```diff
@@ -7962,11 +7966,7 @@ export default class Meeting extends StatelessWebexPlugin {
    *
    * @returns {object} containing timeLeft, expiryTime, currentTime
    */
-  public getPermissionTokenExpiryInfo(): {
-    timeLeft?: number;
-    expiryTime?: number;
-    currentTime: number;
-  } {
+  public getPermissionTokenExpiryInfo() {
```
Comment (marcin-bazyl) [inline packages/@webex/plugin-meetings/src/meeting/index.ts:7969]: wouldn't it be simpler to just change code in line 7971 to `return {};`?

Comment (shnaaz) [reply packages/@webex/plugin-meetings/src/meeting/index.ts:7969]: We decided against that, if there is no payload then that means there is nothing and returning empty object is not ideal

**Review by jor-row — APPROVED** [review]
was feeling left out

**Pull request merged**